plugins/org.jboss.tools.livereload.core/src/org/jboss/tools/livereload/core/internal/util/PathBuilder.java

```diff
@@ -0,0 +1,44 @@
+package org.jboss.tools.livereload.core.internal.util;
+
+/**
+ * Convenient utility class to build URL path and make sure {@code /} separators
+ * are properly handled.
+ */
+public class PathBuilder {
+
+	private final StringBuilder path = new StringBuilder();
+
+	public static PathBuilder from(final String pathFragment) {
+		final PathBuilder pathBuilder = new PathBuilder();
+		return pathBuilder.path(pathFragment);
+	}
+
+	public PathBuilder path(final String pathFragment) {
+		if (pathFragment == null) {
+			return this;
+		}
+		if (!pathFragment.startsWith("/") && !currentPathEndsWith('/')) { 
+			this.path.append("/").append(pathFragment);
+		} else if(!pathFragment.startsWith("/") && currentPathEndsWith('/')) {
```

Comment:
ex. this and the following case are the exact opposite, with grouping this would be much easier to understand
!pathFragment.startsWith("/") && currentPathEndsWith('/')
pathFragment.startsWith("/") && !currentPathEndsWith('/')